Improve draft commit request and response bodies

ffakenz opened the pull request:
<!-- Describe your change here -->
Fixes #945 

* Uses the "TextEnvelope" format on the response so directly usable by cardano-cli
* This also removes the need for wrapping the `UTxO` into `{utxos: ...}` on the request side

---

<!-- Consider each and tick it off one way or the other -->
* [x] CHANGELOG updated or not needed
* [x] Documentation updated or not needed
* [x] Haddocks updated or not needed
* [x] No new TODOs introduced or explained herafter

ffakenz committed:
Refactor json instances for response type so to be a serialized text envelope instead

This allows clients to submit the drafted tx in the response directly to cardano cli.

ch1bo commented on hydra-node/src/Hydra/API/RestServer.hs:
```diff
@@ -38,19 +37,14 @@ newtype DraftCommitTxResponse = DraftCommitTxResponse
-    object
-      [ "commitTx" .= (String . decodeUtf8 . Base16.encode $ serialiseToCBOR tx)
-      ]
+    toJSON $ serialiseToTextEnvelope (Just "Tx Babbage") tx
```
Could: Provide a more descriptive description what kind of transaction this is.
```suggestion
    toJSON $ serialiseToTextEnvelope (Just "Hydra commit transaction") tx
```

ch1bo commented on hydra-node/json-schemas/api.yaml:
```diff
@@ -791,13 +791,14 @@ components:
+        example:
+          {
+              "cborHex": "8303018282051927168200581c0d94e174732ef9aae73f395ab44507bfa983d65023c11a951f0c32e4",
+              "description": "Tx Babbage",
```
Following my other comment, this description example could be updated as well.

ch1bo commented on hydra-node/json-schemas/api.yaml:
```diff
@@ -791,13 +791,14 @@ components:
-            $ref: "#/components/schemas/RawTransaction"
+        example:
+          {
+              "cborHex": "8303018282051927168200581c0d94e174732ef9aae73f395ab44507bfa983d65023c11a951f0c32e4",
```
Is this a valid transaction? seems quite short

ffakenz replied:
not sure, took the cbor example from "#/components/schemas/Script".

anyhow, now the example is being autogenerated and this is no longer needed.

ch1bo commented on hydra-node/json-schemas/api.yaml:
```diff
@@ -791,13 +791,14 @@ components:
-        required:
-          - commitTx
+        $ref: "#/components/schemas/RawTransaction"
         additionalProperties: false
```
Must: fix the schema

This is wrong, it's a `TextEnvelopeTransaction`

```suggestion
        $ref: "#/components/schemas/TextEnvelopeTransaction"
```

ch1bo replied:
Should: also fix the TextEnvelopeTransaction schema which uses still the old `type`

https://github.com/input-output-hk/hydra/blob/e0f011829e60ca409e9c59e23bc0d2f5e0d4f02b/hydra-node/json-schemas/api.yaml#L1374 should say `Tx BabbageEra` as we usually produce and consume babbage era transactions.

ch1bo replied:
Done in https://github.com/input-output-hk/hydra/commit/6dc6a9cecf7036cad47c68e8815ef48925e4da34

ch1bo committed:
Fix api.yaml on /commit types and tests checking it

Unfortunately, prop_validateJSONSchema was failing silently with an
unresolvable "path" into the schema. We should not hand-roll these
things.

ch1bo committed:
Use a more descriptive description

Also removes the example as a full example would necessarily increase
the api spec size by 4k or so.

ch1bo committed:
Make async-api validator resolve paths correctly

As we are now using named self-references, which are required by our
hand-rolled property tests on validating json schema, the async api
validation also needs to know the path of the file to resolve correctly.

ch1bo committed:
DraftCommitTxRequest is a UTxOWithWitness directly

This removes the need of wrapping a UTxO json object in single field
object and simplifies requests to the /commit endpoint.

ffakenz committed:
Fix typo on attribute for utxo to commit in request
